internal/webhook/spinappexecutor_validating.go

```diff
@@ -50,5 +53,21 @@ func (v *SpinAppExecutorValidator) ValidateDelete(ctx context.Context, obj runti
 }
 
 func (v *SpinAppExecutorValidator) validateSpinAppExecutor(executor *spinv1alpha1.SpinAppExecutor) error {
+	var allErrs field.ErrorList
+	if err := validateOpenTelemetry(*executor.Spec.DeploymentConfig); err != nil {
+		allErrs = append(allErrs, err)
+	}
+	if len(allErrs) == 0 {
+		return nil
+	}
+
+	return apierrors.NewInvalid(schema.GroupKind{Group: "core.spinoperator.dev", Kind: "SpinAppExecutor"}, executor.Name, allErrs)
+}
+
+func validateOpenTelemetry(deployment spinv1alpha1.ExecutorDeploymentConfig) *field.Error {
+	if deployment.Otel.Endpoint == "" {
```

Comment:
Do we actually want to make otel a requirement? - probably should be optional

Reply:
+1 to it being optional